Sources/StreamChat/WebSocketClient/EventMiddlewares/ChannelTruncatedEventMiddleware.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+/// The middleware listens for `ChannelTruncatedEventMiddleware` events and updates `ChannelDTO` accordingly.
+struct ChannelTruncatedEventMiddleware<ExtraData: ExtraDataTypes>: EventMiddleware {
+    let database: DatabaseContainer
+
+    func handle(event: Event, completion: @escaping (Event?) -> Void) {
+        guard
+            let truncatedEvent = event as? ChannelTruncatedEvent,
+            let payload = truncatedEvent.payload as? EventPayload<ExtraData>
+        else {
+            completion(event)
+            return
+        }
+
+        database.write { session in
+            if let channelDTO = session.channel(cid: truncatedEvent.cid) {
+                channelDTO.truncatedAt = payload.createdAt
+            } else {
+                throw ClientError.ChannelDoesNotExist(cid: truncatedEvent.cid)
+            }
+        } completion: { error in
+            if let error = error {
+                log.error("Failed to update message reaction in the database, error: \(error)")
```

Comment:
(While copying your code for my Watcher Middleware) I've realized you forgot to change the error message here :trollface:

Reply:
Nice 😅